### PR TITLE
remove output mode handles

### LINKED ISSUE #7304: Build failure
With latest sources:
```
ninja: Entering directory `build'
[363/611] Compiling C object sway/sway.p/lock.c.o
FAILED: sway/sway.p/lock.c.o
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.9-dev-3bd657c7 (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/lock.c.o -MF sway/sway.p/lock.c.o.d -o sway/sway.p/lock.c.o -c ../sway/lock.c
../sway/lock.c: In function ‘handle_new_surface’:
../sway/lock.c:119:50: error: ‘struct <anonymous>’ has no member named ‘mode’
  119 |         wl_signal_add(&output->wlr_output->events.mode, &surf->output_mode);
      |                                                  ^
[376/611] Compiling C object sway/sway.p/desktop_output.c.o
FAILED: sway/sway.p/desktop_output.c.o
cc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Isubprojects/wlroots/include -I../subprojects/wlroots/include -Iprotocols -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/json-c -I/usr/include/libevdev-1.0 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O0 -g -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wundef -Wvla '-DSYSCONFDIR="//usr/local/etc"' '-DSWAY_VERSION="1.9-dev-3bd657c7 (" __DATE__ ", branch '"'"'master'"'"')"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/desktop_output.c.o -MF sway/sway.p/desktop_output.c.o.d -o sway/sway.p/desktop_output.c.o -c ../sway/desktop/output.c
../sway/desktop/output.c: In function ‘handle_new_output’:
../sway/desktop/output.c:957:42: error: ‘struct <anonymous>’ has no member named ‘mode’
  957 |         wl_signal_add(&wlr_output->events.mode, &output->mode);
      |                                          ^
[380/611] Linking target subprojects/wlroots/libwlroots.so.12
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by bl4ckb0ne**
See [1]

Closes #7304 

[1]: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/3911

### CODE REVIEW

File: sway/lock.c [hunk 1]
```diff
@@ -56,6 +49,11 @@ static void handle_output_commit(struct wl_listener *listener, void *data) {
 		wlr_session_lock_surface_v1_configure(surf->lock_surface,
 			surf->output->width, surf->output->height);
 	}
+
+	if (event->committed & WLR_OUTPUT_STATE_MODE) {
+		wlr_session_lock_surface_v1_configure(surf->lock_surface,
```
Comment (emersion) [inline sway/lock.c:54]: This is already handled above

File: sway/desktop/output.c [hunk 2]
```diff
@@ -822,8 +821,7 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 	update_output_manager_config(server);
 }
 
-static void handle_mode(struct wl_listener *listener, void *data) {
-	struct sway_output *output = wl_container_of(listener, output, mode);
+static void output_mode(struct sway_output *output) {
```
Comment (emersion) [inline sway/desktop/output.c:824]: Nit: can we keep the original function name?

Comment (bl4ckb0ne) [reply sway/desktop/output.c]: Of course

File: sway/lock.c (hunk 1 above)
Comment (bl4ckb0ne) [reply sway/lock.c]: Ah yes, should've seen that.

**Review by emersion — APPROVED** [review]
Thanks for the patch!

**Pull request merged**